### DISCUSSION TITLE
Shoelance UI-Components do not get shown properly after error handling.

### DISCUSSION THREAD

**Discussion by stevan06v** [livewire/livewire#8045] [1. Help]
**Problem**
I am using the shoelanceUI-library (via vite) for my UI components and I have stumbled on an unusal error. After loading the component for a second time (in my case after displaying an error message in the UI) the shoelanceUI-component does not get displayed properly (image below).

**Code:**

Livewire-component:

    public function save() {
        $this->validate();

        if (Auth::check()) {

            $this->validate();
            try {
                $project = $this->createProject($this->projectUrl);
                $this->redirect('/projects/' . $project->id);
            } catch (Exception $e) {
                $this->createProjectError = $e->getMessage();
            }
        } else {
            $this->redirect('/register');
        }
    }

Livewire-view:

    @if ($createProjectError)
        <sl-alert variant="danger" duration="3000" closable>
            <sl-icon slot="icon" name="exclamation-octagon"></sl-icon>
            <strong>Project creation-error</strong><br/>
            {{$createProjectError}}
        </sl-alert>
    @endif

Before Error (Properly loaded component):
<img width="947" alt="Bildschirmfoto 2024-02-28 um 22 11 20" src="https://github.com/livewire/livewire/assets/91951603/a95ab005-9de7-4692-bbac-6c2f69ace171">

After error (Not properly loaded component): 

<img width="980" alt="Bildschirmfoto 2024-02-28 um 22 11 30" src="https://github.com/livewire/livewire/assets/91951603/df2be462-ba7d-4fde-a738-46a70fcf3689">

**Comment by danie-ramdhani** [comment]
mmm... Shoelance UI-Components is using javascript to render the `sl-*` tag element to basic html elements.

![image](https://github.com/livewire/livewire/assets/58886605/c453e0eb-f832-4c61-88f6-6b70c304acb8)

livewire is re-render (morph) all the elements in the component when user interact with button with wire:click or live inputs.
it will cause all javascript rendered elemets get wiped, but usually can be prevent by `wire:ignore` (https://livewire.laravel.com/docs/wire-ignore).

example:

```blade
  {{-- ​add wire:ignore attribute to the root element --}}
  <sl-alert variant="primary" open wire:ignore>
      <sl-icon name="info-circle" slot="icon"></sl-icon>
      <strong>This is super informative</strong><br />
      You can tell by how pretty the alert is.
  </sl-alert>
```

idk if it's a good idea to add `wire:ignore` to **ALL** your Shoelance UI-Components. maybe you can try other ui components.

EDIT

if you want to try other ui component for livewire:

- https://devdojo.com/pines
- https://mary-ui.com/

**Reply by stevan06v** [reply]
Thank you that worked for me! 

But I still have problems with such components:

    <sl-alert open wire:ignore>
        <sl-icon slot="icon" name="info-circle"></sl-icon>
        This is a standard alert. You can customize its content and even the icon.
    </sl-alert>

In this case the style also does not get loaded. 

Can you also help me with that :) ?
